src/gtk/textctrl.cpp

```diff
@@ -728,7 +728,26 @@ bool wxTextCtrl::Create( wxWindow *parent,
         gtk_entry_get_text((GtkEntry*)m_text);
 
         if (style & wxNO_BORDER)
-            g_object_set (m_text, "has-frame", FALSE, NULL);
+        {
+#ifdef __WXGTK3__
+            // this is sort of a workaround for when the builtin theme
+            // won't remove the frame by itself -- see
+            // https://bugzilla.gnome.org/show_bug.cgi?id=789732#c1
+            GtkCssProvider* provider = gtk_css_provider_new();
```

Comment:
Could we reuse the existing (but, curiously, currently unused) `wxWindowGTK::ApplyCssStyle()`?